libsolidity/analysis/ControlFlowAnalyzer.cpp

```diff
@@ -144,12 +144,12 @@ void ControlFlowAnalyzer::checkUnassignedStorageReturnValues(
 						ssl.append("Problematic end of function:", _function.location());
 				}
 
-			m_errorReporter.warning(
+			m_errorReporter.typeError(
 				returnVal->location(),
+				ssl,
 				"This variable is of storage pointer type and might be returned without assignment. "
 				"This can cause storage corruption. Assign the variable (potentially from itself) "
```

Comment:
Since it is an error now, it actually cannot cause storage corruption anymore. Perhaps rather phrase it more abstract like `could be used uninitialized` or something like that.